### PR TITLE
Upgrade django based on dependabot security alerts

### PR DESCRIPTION
**Pull request by tznamena**
This PR upgrades Django to address following dependabot alerts:

**Critical:**
https://github.com/ansible/django-ansible-base/security/dependabot/21
**High:**
https://github.com/ansible/django-ansible-base/security/dependabot/18
https://github.com/ansible/django-ansible-base/security/dependabot/16
https://github.com/ansible/django-ansible-base/security/dependabot/15
**Moderate:**
https://github.com/ansible/django-ansible-base/security/dependabot/17
https://github.com/ansible/django-ansible-base/security/dependabot/22
https://github.com/ansible/django-ansible-base/security/dependabot/20
https://github.com/ansible/django-ansible-base/security/dependabot/19

### CODE REVIEW

**Comment by tznamena** [comment]
Keeping this in draft until productization have the 4.2.16 built, its only been released today but we might have the RPM tomorrow.

File: requirements/requirements.in [hunk 1]
```diff
@@ -3,7 +3,7 @@
 # if you are add a new feature which requires dependencies they should be in a separate requirements_<feature>.in file
 #
 cryptography
-Django>=4.2.5,<4.3.0
+Django>=4.2.15,<4.3.0 # pinned to version productization already has available
```
Comment (AlanCoding) [inline requirements/requirements.in]: https://github.com/ansible/awx/blob/devel/requirements/requirements.txt is still on 4.2.10

https://github.com/ansible/awx/pull/15360 was raised (similar reasons), but only went to 4.2.14, which would still be a conflict.

Comment (tznamena) [reply requirements/requirements.in]: Can you please explain why is AWX with different version of Django a conflict for DAB?

Comment (tznamena) [reply requirements/requirements.in]: I had a PR up in AWX and was told that we should not upgrade it there and that we should upgrade it in tower instead. So there is a PR for tower now: https://github.com/ansible/tower/pull/6704

Comment (tznamena) [reply requirements/requirements.in]: The PR in tower has been merged.

**Review by simaishi — APPROVED** [review]

**Pull request merged**